readthedocs/projects/tasks/builds.py

```diff
@@ -480,7 +480,7 @@ def on_success(self, retval, task_id, args, kwargs):
                 api_v2.version(self.data.version.pk).patch(
                     {
                         "built": True,
-                        "documentation_type": self.data.version.documentation_type,
+                        "documentation_type": self.data.config.doctype,
```

Comment:
the previous logic was correct, the real doctype for mkdocs is known at build time

https://github.com/readthedocs/readthedocs.org/blob/9cf20d0e6b05f091a41ac4278622797c692089d8/readthedocs/doc_builder/director.py#L357-L357